src/QirRuntime/samples/StandaloneInputReference/qir-driver.cpp

```diff
@@ -0,0 +1,269 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#include <cstring> // for memcpy
+#include <fstream>
+#include <iostream>
+#include <map>
+#include <vector>
+
+#include "CoreTypes.hpp"
+#include "QirContext.hpp"
+#include "QirTypes.hpp"
+#include "QuantumApi_I.hpp"
+#include "SimFactory.hpp"
+
+#include "quantum__qis_internal.hpp"
+#include "quantum__rt.hpp"
+
+#include "CLI11.hpp"
+
+using namespace Microsoft::Quantum;
+using namespace std;
+
+// This is the function corresponding to the QIR entry-point.
+extern "C" int64_t Quantum__StandaloneSupportedInputs__ExerciseInputs__body( // NOLINT
+    int64_t intValue,
+    double doubleValue,
+    Result resultValue,
+    QirString* stringValue,
+    int64_t rangeStart,
+    int64_t rangeStep,
+    int64_t rangeEnd);
```

Comment:
(Up to you)
I'm comparing this C/C++ signature, the _range_ part
> Quantum__StandaloneSupportedInputs__ExerciseInputs__body(
    . . .
    **int64_t rangeStart,
    int64_t rangeStep,
    int64_t rangeEnd**);

with the `.qs` signature 
> operation ExerciseInputs (. . ., **rangeValue : Range**)

and with the generated `.ll` output
> %Range = type **{ i64, i64, i64 }**
define i64 @Quantum__StandaloneSupportedInputs__ExerciseInputs__body(. . ., **%Range %rangeValue**)

The C/C++ signature looks somewhat _hacky_ ;-)
Would be nice to see the comments in the code explaining that the C/C++ signature is a conscious choice and is compatible with `.ll` (I didn't dive deep but in those comments I would welcome some analysis and mentioning of liknks like these - 
https://llvm.org/docs/LangRef.html#structure-types , 
https://llvm.org/docs/LangRef.html#preallocated-operand-bundles
https://llvm.org/docs/LangRef.html#structure-type
).

The following C/C++ signature seems to me somewhat safer (if it doesn't work then would be nice to see in the comments the explanation why it does not work)
> Quantum__StandaloneSupportedInputs__ExerciseInputs__body(
    . . .
    **QirRange range**);

Reply:
I looked more into this and provided some detailed analysis offline. In summary, it seems like the expectations for a QIR signature like the following:
```
define i64 @Quantum__StandaloneSupportedInputs__ExerciseInputs__body(i64 %intValue, double %doubleValue, %Result* %resultValue, %String* %stringValue, %Range %rangeValue)
```
Where `%Range` is defined as:
```
%Range = type { i64, i64, i64 }
```
Align with the expectations of a C signature like the following:
```
extern "C" int64_t Quantum__StandaloneSupportedInputs__ExerciseInputs__body( // NOLINT
 int64_t intValue,
 double doubleValue,
 Result resultValue,
 QirString* stringValue,
 int64_t rangeStart,
 int64_t rangeStep,
 int64_t rangeEnd);
```
But do not align with the expectations of a C signature like the following:
```
extern "C" int64_t Quantum__StandaloneSupportedInputs__ExerciseInputs__body( // NOLINT
 int64_t intValue,
 double doubleValue,
 Result resultValue,
 QirString* stringValue,
 QirRange rangeValue);
```
I changed the entry-point operation to not receive a Range for now, and I'll suggest to change QIR generation in this case to receive a pointer to QirRange rather than passing it by value since pointers are maybe the most reliable way to interop between C++ and QIR for struct values.

Reply:
We need to expand the arguments received by the QIR operation anyway when the latest compiler entry-point mapping is merged so we can remove this hacky part for now to not confuse people.

Reply:
OK, thank you for the offline analysis, Cesar.
Looks like compiler operates not exactly the way I expected.